### PR TITLE
Add datapane and dacite recipes

### PR DESCRIPTION
**Pull request by mands**
#### Packages
- dacite (https://github.com/konradhalas/dacite)
- datapane (https://github.com/datapane/datapane)

#### Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/datapane/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+requirements:
+  host:
+    - pip
+    - python
+    - setuptools
+    - poetry
+  run:
+    - python
+    - dataclasses  # [py==36]
```
Comment (marcelotrevisani) [inline recipes/datapane/meta.yaml:29]:
```suggestion
    - dataclasses
```

File: recipes/datapane/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  version: {{ version }}
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: 398fdf46259e6e7485e01ddb817fe27158797120346464fbe476c80ac13f8e5b
+  
+build:
+  noarch: python
+  number: 0
+  skip: True  # [py<36]
```
Comment (marcelotrevisani) [inline recipes/datapane/meta.yaml:15]:
```suggestion
```

File: recipes/dacite/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  noarch: python
+  number: 0
+  skip: True  # [py<36]
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - pip
+    - python
+    - dataclasses  # [py==36]
```
Comment (marcelotrevisani) [inline recipes/dacite/meta.yaml:22]:
```suggestion
```

File: recipes/dacite/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: 764c96e0304cb189628686689a163a6a3a8ce7bf3465f0a2d882a8b42f88108f
+
+build:
+  noarch: python
+  number: 0
+  skip: True  # [py<36]
```
Comment (marcelotrevisani) [inline recipes/dacite/meta.yaml:15]:
```suggestion
```

File: recipes/dacite/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - pip
+    - python
+    - dataclasses  # [py==36]
+  run:
+    - python
+    - dataclasses  # [py==36]
```
Comment (marcelotrevisani) [inline recipes/dacite/meta.yaml:25]:
```suggestion
    - dataclasses
```

File: recipes/dacite/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    - dacite
+
+about:
+  home: "https://github.com/konradhalas/dacite"
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE
+  summary: "Simple creation of data classes from dictionaries."
+  doc_url: 
+  dev_url:
```
Comment (marcelotrevisani) [inline recipes/dacite/meta.yaml:38]:
```suggestion
```

File: recipes/dacite/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    - pip
+    - python
+    - dataclasses  # [py==36]
+  run:
+    - python
+    - dataclasses  # [py==36]
+
+test:
+  imports:
+    - dacite
```
Comment (marcelotrevisani) [inline recipes/dacite/meta.yaml:29]: Can we add ``pip check`` here please?
```suggestion
    - dacite
  requires:
    - pip
  commands:
    - pip check
```

File: recipes/datapane/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+  commands:
+    - datapane --help
+    - datapane --version
+    - dp-runner --help
+
+about:
+  home: https://datapane.com
+  license: Proprietary
+  license_family: PROPRIETARY
+  license_file: {{ name }-{{ version }}/LICENSE.txt
```
Comment (marcelotrevisani) [inline recipes/datapane/meta.yaml:78]:
```suggestion
  license_file: LICENSE.txt
```

File: recipes/datapane/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+    - datapane.client.scripts
+    - datapane.common
+    - datapane.resources
+    - datapane.resources.local_report
+    - datapane.resources.scaffold
+    - datapane.runner
+  commands:
+    - datapane --help
+    - datapane --version
+    - dp-runner --help
```
Comment (marcelotrevisani) [inline recipes/datapane/meta.yaml:72]:
```suggestion
    - dp-runner --help
    - pip check
  requires:
    - pip
```

**Comment by mands** [comment]
Thank you for the feedback - I'll endeavour to get it cleaned up and updated over the weekend.

File: recipes/datapane/meta.yaml (hunk 1 above)
Comment (mands) [reply recipes/datapane/meta.yaml:29]: `dataclasses` is built into the standard library on python 3.7+ - will this cause a problem?

File: recipes/datapane/meta.yaml (hunk 2 above)
Comment (mands) [reply recipes/datapane/meta.yaml:15]: What's the best way to specify in the recipe that the package is pure-Python but only supports python 3.6 and above ?

File: recipes/dacite/meta.yaml (hunk 7 above)
Comment (mands) [reply recipes/dacite/meta.yaml:29]: Cool - will do

File: recipes/datapane/meta.yaml (hunk 8 above)
Comment (mands) [reply recipes/datapane/meta.yaml:78]: This LICENSE file is bundled with the `sdist` in the location indicated by the template above - will conda-build find it in the nested subdirectory?

**Comment by mands** [comment]
@conda-forge-admin, please ping team

^ just added a few feedback questions to the review on some things I'm unsure about

File: recipes/datapane/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+  requires:
+    - pip
+  commands:
+    - pip check
+
+about:
+  home: https://datapane.com
+  license: Proprietary
```
Comment (chrisburr) [inline recipes/datapane/meta.yaml:78]: We can't distribute this a the license file in the tarball states that you can't do anything with this package:
```
Copyright (c) 2015-2020, StackHut Limited (trading as Datapane).
All rights reserved.
```
At the very least conda-forge, NumFOCUS and Anaconda Inc need to have permission to redistribute the package without restrictions.

Comment (mands) [reply recipes/datapane/meta.yaml]: Thanks for the feedback. No problem - we were in the process of moving to Apache-2.0 internally anyway so I just implemented that over the weekend and repushed - cheers!

**Comment by mands** [comment]
@conda-forge-admin, please ping team

requesting a review on the latest feedback changes

File: recipes/datapane/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+    - click >=7.0.0,<8.0.0
+    - click-spinner >=0.1.8,<0.2.0
+    - colorlog >=4.1.0,<5.0.0
+    - dacite >=1.5.0,<2.0.0
+    - flit-core >=2.3.0,<2.4.0
+    - folium >=0.11.0,<0.12.0
+    - importlib_resources >=3.0.0,<4.0.0
+    - jinja2 >=2.11.1,<3.0.0
+    - jsonschema >=3.2.0,<4.0.0
+    - matplotlib >=3.1.0,<4.0.0
```
Comment (chrisburr) [inline recipes/datapane/meta.yaml:40]: If you don't need the qt frontend ([more details](https://conda-forge.org/docs/maintainer/knowledge_base.html#matplotlib)):
```suggestion
    - matplotlib-base >=3.1.0,<4.0.0
```

Comment (mands) [reply recipes/datapane/meta.yaml:40]: Got it - will update and repush

**Comment by mands** [comment]
@conda-forge-admin, please ping team

requesting a review on the latest feedback changes

**Comment by mands** [comment]
Great - thanks!

**Pull request merged**